### PR TITLE
Revise building readme

### LINKED ISSUE #5701: Readme cleanups for 2.0
Let's make a few improvements to the readmes that we can merge at the same time as we do the 2.0 release:
* the "modules" section is really confusingly named to me... let's rename it to something like "partial bundles" and then highlight that there are two kinds: official/distributed ones (covered in the dist readme) and custom ones (covered in a new readme that will document the custom process)
* the "building" section should encourage folks to use a partial bundle first, and if not, then basically we should describe how to build the library, although the use-case there is basically (here is what you'll need to do to build *your* library if your library imports some modules in some other way than via a custom bundle)
* the dist readme should not contain anything pertaining to MathJax (it used to, because we used to distribute MathJax but we don't any more) and instead MathJax/LaTeX-related stuff should be a short new section in the main readme

### PR DESCRIPTION
**Pull request by archmoj**
Closes #5701 by addressing part 2 of the issue.

@plotly/plotly_js

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,88 +72,119 @@ and use the plotly.js `dist` file(s). More info [here](https://github.com/plotly
...
+
+By default, plotly.js will modify the global MathJax configuration on load.
+This can lead to undesirable behavior if plotly.js is loaded alongside
+other libraries that also rely on MathJax. To disable this global configuration
+process, set the `MathJaxConfig` property to `'local'` in the `window.PlotlyConfig`
+object.  This property must be set before the plotly.js script tag, for example:
+
+```html
+<script>
+   window.PlotlyConfig = {MathJaxConfig: 'local'}
```
Comment (alexcjohnson) [inline README.md:161]: `PlotlyConfig` is no longer a thing in v2, so can't we remove all of this? Also does the script tag order even matter anymore? I bet it works now before or after, though I haven't tested this myself.

Comment (archmoj) [reply README.md]: Good call. Revisited in https://github.com/plotly/plotly.js/pull/5704/commits/5c6f724a980bbfe2ab5ba77405d9dd049fa60dad.

File: BUILDING.md [hunk 2]
```diff
@@ -1,6 +1,24 @@
...
+
+then simply run
+
+```sh
+browserify index.js > bundle.js
+```
+
+
+
+The section below provide additional info in respect to alternative building frameworks.
```
Comment (alexcjohnson) [inline BUILDING.md:21]:
```suggestion
The sections below provide additional info in respect to alternative building frameworks.
```

**Review by alexcjohnson — APPROVED** [review]
💃

**Pull request merged**